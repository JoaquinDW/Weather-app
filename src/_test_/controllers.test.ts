import { expect, test, describe } from "vitest";
import {
  KelvinToCelcius,
  getWeatherData,
  formatCurrentWeather,
} from "../Services/controllers";

describe("kelvinToCelcius", () => {
  test("should convert temperatures from Kelvin to Celcius", () => {
    expect(KelvinToCelcius(0)).toBe(-273);
    expect(KelvinToCelcius(300)).toBe(27);
  });
});

describe("getWeatherData", () => {
  test("should get weather data", async () => {
    const data = await getWeatherData("weather", { q: "London" });
    expect(data).toHaveProperty("coord");
  });
  test("should get forecast data", async () => {
    const data = await getWeatherData("forecast", { q: "London" });
    expect(data).toHaveProperty("list");
  });
});

describe("formatCurrentWeather", () => {
  test("should format current weather data", async () => {
    const data = await getWeatherData("weather", { q: "London" });
    const formatedData = formatCurrentWeather(data);
    expect(formatedData).toHaveProperty("name");
    expect(formatedData).toHaveProperty("dt");
    expect(formatedData).toHaveProperty("country");
    expect(formatedData).toHaveProperty("sunrise");
    expect(formatedData).toHaveProperty("sunset");
    expect(formatedData).toHaveProperty("details");
    expect(formatedData).toHaveProperty("icon");
    expect(formatedData).toHaveProperty("speed");
  });
});
