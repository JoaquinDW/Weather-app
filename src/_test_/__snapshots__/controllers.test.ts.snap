// Vitest Snapshot v1

exports[`getWeatherData > should get weather data > https://api.openweathermap.org/data/2.5/weather?q=London&apasdsasdasddpid=923c12a56b9f714ede562b89e7c9523f 1`] = `
{
  "base": "stations",
  "clouds": {
    "all": 0,
  },
  "cod": 200,
  "coord": {
    "lat": 51.5085,
    "lon": -0.1257,
  },
  "dt": 1666063123,
  "id": 2643743,
  "main": {
    "feels_like": 282.38,
    "humidity": 94,
    "pressure": 1025,
    "temp": 282.38,
    "temp_max": 284.01,
    "temp_min": 278.21,
  },
  "name": "London",
  "sys": {
    "country": "GB",
    "id": 2075535,
    "sunrise": 1666074545,
    "sunset": 1666112517,
    "type": 2,
  },
  "timezone": 3600,
  "visibility": 10000,
  "weather": [
    {
      "description": "clear sky",
      "icon": "01n",
      "id": 800,
      "main": "Clear",
    },
  ],
  "wind": {
    "deg": 0,
    "speed": 1.03,
  },
}
`;

exports[`getWeatherData > should get weather data > https://api.openweathermap.org/data/2.5/weather?q=London&apasdsdpid=923c12a56b9f714ede562b89e7c9523f 1`] = `
{
  "base": "stations",
  "clouds": {
    "all": 0,
  },
  "cod": 200,
  "coord": {
    "lat": 51.5085,
    "lon": -0.1257,
  },
  "dt": 1666063123,
  "id": 2643743,
  "main": {
    "feels_like": 282.38,
    "humidity": 94,
    "pressure": 1025,
    "temp": 282.38,
    "temp_max": 284.01,
    "temp_min": 278.21,
  },
  "name": "London",
  "sys": {
    "country": "GB",
    "id": 2075535,
    "sunrise": 1666074545,
    "sunset": 1666112517,
    "type": 2,
  },
  "timezone": 3600,
  "visibility": 10000,
  "weather": [
    {
      "description": "clear sky",
      "icon": "01n",
      "id": 800,
      "main": "Clear",
    },
  ],
  "wind": {
    "deg": 0,
    "speed": 1.03,
  },
}
`;

exports[`getWeatherData > should get weather data > https://api.openweathermap.org/data/2.5/weather?q=London&appid=923c12a56b9f714ede562b89e7c9523f 1`] = `
{
  "base": "stations",
  "clouds": {
    "all": 0,
  },
  "cod": 200,
  "coord": {
    "lat": 51.5085,
    "lon": -0.1257,
  },
  "dt": 1666063123,
  "id": 2643743,
  "main": {
    "feels_like": 282.38,
    "humidity": 94,
    "pressure": 1025,
    "temp": 282.38,
    "temp_max": 284.01,
    "temp_min": 278.21,
  },
  "name": "London",
  "sys": {
    "country": "GB",
    "id": 2075535,
    "sunrise": 1666074545,
    "sunset": 1666112517,
    "type": 2,
  },
  "timezone": 3600,
  "visibility": 10000,
  "weather": [
    {
      "description": "clear sky",
      "icon": "01n",
      "id": 800,
      "main": "Clear",
    },
  ],
  "wind": {
    "deg": 0,
    "speed": 1.03,
  },
}
`;
